install/kots/manifests/gitpod-installer-job.yaml

```diff
@@ -273,6 +272,26 @@ spec:
                 'del(select(.kind == "StatefulSet" and .metadata.name == "openvsx-proxy").status)' \
                 "${GITPOD_OBJECTS}/templates/gitpod.yaml"
 
+              # Merge different docker pull secrets into one secret
+              if [ '{{repl HasLocalRegistry }}' = "true" ];
+              then
+                kubectl get secret "{{repl ImagePullSecretName }}" -o=jsonpath="{.data['\.dockerconfigjson']}" | base64 --decode | yq -P - > registry-auth-airgap.yaml
+              fi
+              if [ '{{repl ConfigOptionEquals "reg_incluster" "0" }}' = "true" ];
+              then
+                kubectl get secret "container-registry" -o=jsonpath="{.data['\.dockerconfigjson']}" | base64 --decode | yq -P - > registry-auth-external.yaml
+              else
+                yq eval 'select(.kind == "Secret" and .metadata.name == "builtin-registry-auth").data.".dockerconfigjson"' \
+                  "${GITPOD_OBJECTS}/templates/gitpod.yaml" | base64 --decode | yq -P - > registry-auth-builtin.yaml
+              fi
+              # merge all files together (https://mikefarah.gitbook.io/yq/operators/reduce#merge-all-yaml-files-together)
+              yq -o=json eval-all '. as $item ireduce ({}; . * $item )' registry-auth-*.yaml > merged-registry-auths.yaml
+              # create secret and update if exists (https://stackoverflow.com/a/45881259/1364435)
+              kubectl create secret generic merged-registry-auths \
```

Comment:
We could also create a secret manifest file and move it to the helm templates directory.